### PR TITLE
fix(other): jmap quota display

### PR DESCRIPTION
**Pull request by Shadow243**
RFC: [https://www.rfc-editor.org/rfc/rfc9425.pdf](https://www.rfc-editor.org/rfc/rfc9425.pdf)
Add `get_quota_root` to hm-jmap.php to correct error `Uncaught Error: Call to undefined method Hm_JMAP::get_quota_root`
Screenshot 2024-10-22 at 18 15 28

### CODE REVIEW

**Comment by Shadow243** [comment]
There an other issue that i'l correct: `Uncaught Error: Call to undefined method Hm_JMAP::get_capability()`

**Comment by Shadow243** [comment]
> There an other issue that i'l correct: `Uncaught Error: Call to undefined method Hm_JMAP::get_capability()`

I just added `get_capability` that must be implemented, I just spent several hours on this but no method worked, so I have to look for another way.

File: modules/imap/handler_modules.php [hunk 1]
```diff
@@ -758,14 +758,32 @@ public function process() {
...
                 $quota_root = $imap->get_quota_root($folder ? $folder : 'INBOX');
-                if ($quota_root && isset($quota_root[0]['name'])) {
-                    $quota = $imap->get_quota($quota_root[0]['name']);
-                    if ($quota) {
-                        $current = floatval($quota[0]['current']);
-                        $max = floatval($quota[0]['max']);
-                        if ($max > 0) {
-                            $this->out('quota', ceil(($current / $max) * 100));
-                            $this->out('quota_max', $max / 1024);
+                if($imap instanceof Hm_JMAP) {
```
[the remark this answers is not in the document]
Comment (Shadow243) [reply modules/imap/handler_modules.php]: @kroky I updated

**Comment by marclaporte** [comment]
@kroky all good?

**Pull request merged**